src/Terrain/TerrainType.java

```diff
@@ -0,0 +1,123 @@
+package Terrain;
+
+import java.awt.Color;
+
+public class TerrainType
+{
+  // Local class data
+  private int mDefenseLevel = -1; // Level of protection provided by this terrain type. Typically 0-4.
+  private  Color mMainColor = null; // Predominant color of this terrain type. Here for convenience.
+  private int mAttributes = 0; // bitmask of binary tile characteristics.
+
+  // Generic constructor.
+  private TerrainType(int attributeFlags, int defense, Color mainColor)
+  {
+    mAttributes = attributeFlags;
+    mDefenseLevel = defense;
+    mMainColor = mainColor;
+  }
+  
+  ////////////////////////////////////////////////////////////////////////////////////////////////////////
+  //// Protected constants to initialize/parse the attribute flags, and the methods to query them.
+  private static final int LAND = 1;
+  private static final int WATER = LAND << 1;
+  private static final int CAPTURABLE = WATER << 1; // Whether a Commander can take ownership of this property.
+  private static final int PROFITABLE = CAPTURABLE << 1; // Whether this terrain type grants income to its owner each turn.
```

Comment:
I think it may be worth supporting a scaling factor on property income.
For instance, having "oil rigs" be more valuable than normal properties might actually entice people to use sea units.
Likewise, having more-valuable properties on land would allow a higher level of funding with less time spent capturing, and less map space devoted to the quite-defensive urban terrain.

Reply:
How would you capture oil rigs? Maybe just ensure there is always shoal nearby? It's an interesting idea, anyway.
We can always have income scaling be a game attribute, but I assume you mean allowing different properties to provide different amounts of income. I'm not necessarily opposed (maybe add mines along with oil rigs?), but I think that should be a future change, if/when we decide we want to add that complexity.

Reply:
Yes, I meant for certain properties to be more valuable (or even less valuable) than others.
My thought was oil rigs would give 3x income, but only while actively occupied by a sea unit.
You wouldn't capture them, and you'd need at least 2 vessels to guarantee income every turn, and likely an APC... but the sea units would also hopefully be useful on their own, so it's not just a sunk cost thing. (This was in my earlier sea-combat pitch, which I will likely revisit once I'm satisfied with the damage system)
Mines... If you mean the places you dig for metal and stuff, I guess they could grant extra income but not heal?
If you mean the kind that explode when you walk over them, uhhh, that'd be interesting...

Reply:
Why would you need two ships to guarantee income every turn? And just... any ship would do?
I'd potentially be open to introducing either type of mine at some point, but I think we should nail down the base game mechanics before we start adding new features.

Reply:
You need two ships because ships passively run out of fuel, and APCs can't swim.
I was thinking any ship would do; you're likely to use the less expensive ones, but IMO there's no reason to bar the more expensive ones. It makes intuitive sense that a footsoldier can do more with urban areas than a tank... but ships are just ships.
Fair enough.